add-on/src/popup/quick-upload.js

```diff
@@ -81,20 +87,36 @@ async function processFiles (state, emitter, files) {
       if (partialResponse) {
         throw new Error('Result of ipfs.add call is missing entries. This may be due to a bug in HTTP API similar to https://github.com/ipfs/go-ipfs/issues/5168')
       }
-      await ipfsCompanion.uploadResultHandler({ result, openRootInNewTab: true })
+
+      // cp will fail if directory does not exist
+      await ipfsCompanion.ipfs.files.mkdir(`${uploadDir}`, { parents: true })
+      // remove directory from files API import files
+      let files = result.filter(file => (file.path !== ''))
+      files = files.map(file => (ipfsCompanion.ipfs.files.cp(`/ipfs/${file.hash}`, `${uploadDir}${file.path}`)))
+      await Promise.all(files)
     } catch (err) {
-      console.error('Failed to IPFS add', err)
+      console.error('Failed to import files to IPFS', err)
       ipfsCompanion.notify('notify_uploadErrorTitle', 'notify_inlineErrorMsg', `${err.message}`)
       throw err
     }
+    state.progress = 'Completed'
     emitter.emit('render')
-    console.log('Upload result', result)
+    console.log(`Successfully imported ${streams.length} files`)
+
+    // open web UI at proper directory
+    // unless and embedded node is in use (no access to web UI)
+    // in which case, open resource.
+    if (state.ipfsNodeType === 'embedded' || !state.openViaWebUI) {
+      await ipfsCompanion.uploadResultHandler({ result, openRootInNewTab: true })
```

Comment:
Noticed this is not working as expected for me tonight. Maybe it's some weirdness with my connection; I'll take a look at this tomorrow.

Reply:
Same issue this morning, importing multiple files and opening them in a context menu hangs. However, it is reproducible on current master, so I won't deal with it in this PR.